<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>642</width>
    <height>553</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TCP调试工具</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="tcpServerLabel">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>5</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Aldhabi</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>服务器</string>
    </property>
   </widget>
   <widget class="QPushButton" name="listeningTrig">
    <property name="geometry">
     <rect>
      <x>220</x>
      <y>55</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>侦听</string>
    </property>
   </widget>
   <widget class="QSplitter" name="splitter">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>55</y>
      <width>183</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <widget class="QSplitter" name="splitter_2">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QLabel" name="port_label">
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="text">
       <string>侦听端口：</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="listenPort"/>
    </widget>
   </widget>
   <widget class="QSplitter" name="splitter_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>85</y>
      <width>131</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <widget class="QLabel" name="maxNum_label">
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>最大连接数：</string>
     </property>
    </widget>
    <widget class="QSpinBox" name="maxConNum">
     <property name="maximum">
      <number>1000</number>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="ip_port_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>115</y>
      <width>111</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>客户端ip及端口号</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="rcvDataDisplay">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>185</y>
      <width>281</width>
      <height>121</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QComboBox" name="ipAndPortsDisplay">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>135</y>
      <width>281</width>
      <height>22</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="refuseConnTrig">
    <property name="geometry">
     <rect>
      <x>140</x>
      <y>110</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>断开</string>
    </property>
   </widget>
   <widget class="QLabel" name="rcvData_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>165</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>接收数据</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="rcvTextSelect">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>165</y>
      <width>51</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>文本</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">rcvTextOrHex</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="rcvHexSelect">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>165</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>十六进制</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">rcvTextOrHex</string>
    </attribute>
   </widget>
   <widget class="QLabel" name="sndData_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>315</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>发送数据</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="sndHexSelect">
    <property name="geometry">
     <rect>
      <x>190</x>
      <y>315</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>十六进制</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">sndTextOrHex</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="sndTextSelect">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>315</y>
      <width>51</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>文本</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">sndTextOrHex</string>
    </attribute>
   </widget>
   <widget class="QTextEdit" name="sndDataprepare">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>335</y>
      <width>281</width>
      <height>121</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="sndDataTrig">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>310</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>发送</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearRcvDataTrig">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>160</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>清除</string>
    </property>
   </widget>
   <widget class="QLabel" name="rcvFile_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>470</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>接收文件</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="rcvFileName">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>470</y>
      <width>171</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="sndFileName">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>510</y>
      <width>171</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="sndFile_label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>510</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>发送文件</string>
    </property>
   </widget>
   <widget class="QPushButton" name="sndFileTrig">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>510</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>发送</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="rcvFileSelect">
    <property name="geometry">
     <rect>
      <x>260</x>
      <y>470</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>接收</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearSndDataTrig">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>310</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>清除</string>
    </property>
   </widget>
   <widget class="QLabel" name="tcpClientLabel">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>10</y>
      <width>71</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Aldhabi</family>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>客户端</string>
    </property>
   </widget>
   <widget class="QLabel" name="port_label_client">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>80</y>
      <width>65</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>连接端口：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="externalServerPort">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>80</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="connectToExternalServerTrig">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>80</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>连接</string>
    </property>
   </widget>
   <widget class="QLabel" name="ip_label_client">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>55</y>
      <width>81</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>连接IP地址：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="externalServerIpAddr">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>55</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="externalServerInfo_label">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>110</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>已连接到服务器：</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="externalServerInfo">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>110</y>
      <width>171</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="rcvHexSelect_client">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>165</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>十六进制</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">rcvTextOrHex_client</string>
    </attribute>
   </widget>
   <widget class="QLabel" name="rcvData_label_client">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>165</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>接收数据</string>
    </property>
   </widget>
   <widget class="QCheckBox" name="rcvFileSelect_client">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>470</y>
      <width>51</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>接收</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearRcvDataTrig_client">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>160</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>清除</string>
    </property>
   </widget>
   <widget class="QLabel" name="rcvFile_label_client">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>470</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>接收文件</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="rcvFileName_client">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>470</y>
      <width>171</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="sndData_label_client">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>315</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>发送数据</string>
    </property>
   </widget>
   <widget class="QPushButton" name="sndDataTrig_client">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>310</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>发送</string>
    </property>
   </widget>
   <widget class="QPushButton" name="clearSndDataTrig_client">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>310</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>清除</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="rcvDataDisplay_client">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>185</y>
      <width>281</width>
      <height>121</height>
     </rect>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QRadioButton" name="sndHexSelect_client">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>315</y>
      <width>71</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>十六进制</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">sndTextOrHex_client</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="rcvTextSelect_client">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>165</y>
      <width>51</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>文本</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">rcvTextOrHex_client</string>
    </attribute>
   </widget>
   <widget class="QTextEdit" name="sndDataprepare_client">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>335</y>
      <width>281</width>
      <height>121</height>
     </rect>
    </property>
   </widget>
   <widget class="QRadioButton" name="sndTextSelect_client">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>315</y>
      <width>51</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>文本</string>
    </property>
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">sndTextOrHex_client</string>
    </attribute>
   </widget>
   <widget class="QLineEdit" name="sndFileName_client">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>510</y>
      <width>171</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="sndFile_label_client">
    <property name="geometry">
     <rect>
      <x>340</x>
      <y>510</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>发送文件</string>
    </property>
   </widget>
   <widget class="QPushButton" name="sndFileTrig_client">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>510</y>
      <width>41</width>
      <height>23</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>10</pointsize>
     </font>
    </property>
    <property name="text">
     <string>发送</string>
    </property>
   </widget>
   <zorder>splitter_3</zorder>
   <zorder>tcpServerLabel</zorder>
   <zorder>listeningTrig</zorder>
   <zorder>splitter</zorder>
   <zorder>ip_port_label</zorder>
   <zorder>rcvDataDisplay</zorder>
   <zorder>ipAndPortsDisplay</zorder>
   <zorder>refuseConnTrig</zorder>
   <zorder>rcvData_label</zorder>
   <zorder>rcvTextSelect</zorder>
   <zorder>rcvHexSelect</zorder>
   <zorder>sndData_label</zorder>
   <zorder>sndHexSelect</zorder>
   <zorder>sndTextSelect</zorder>
   <zorder>sndDataprepare</zorder>
   <zorder>sndDataTrig</zorder>
   <zorder>clearRcvDataTrig</zorder>
   <zorder>rcvFile_label</zorder>
   <zorder>rcvFileName</zorder>
   <zorder>sndFileName</zorder>
   <zorder>sndFile_label</zorder>
   <zorder>sndFileTrig</zorder>
   <zorder>rcvFileSelect</zorder>
   <zorder>clearSndDataTrig</zorder>
   <zorder>tcpClientLabel</zorder>
   <zorder>port_label_client</zorder>
   <zorder>externalServerPort</zorder>
   <zorder>connectToExternalServerTrig</zorder>
   <zorder>ip_label_client</zorder>
   <zorder>externalServerIpAddr</zorder>
   <zorder>externalServerInfo_label</zorder>
   <zorder>externalServerInfo</zorder>
   <zorder>rcvHexSelect_client</zorder>
   <zorder>rcvData_label_client</zorder>
   <zorder>rcvFileSelect_client</zorder>
   <zorder>clearRcvDataTrig_client</zorder>
   <zorder>rcvFile_label_client</zorder>
   <zorder>rcvFileName_client</zorder>
   <zorder>sndData_label_client</zorder>
   <zorder>sndDataTrig_client</zorder>
   <zorder>clearSndDataTrig_client</zorder>
   <zorder>rcvDataDisplay_client</zorder>
   <zorder>sndHexSelect_client</zorder>
   <zorder>rcvTextSelect_client</zorder>
   <zorder>sndDataprepare_client</zorder>
   <zorder>sndTextSelect_client</zorder>
   <zorder>sndFileName_client</zorder>
   <zorder>sndFile_label_client</zorder>
   <zorder>sndFileTrig_client</zorder>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="rcvTextOrHex"/>
  <buttongroup name="sndTextOrHex"/>
  <buttongroup name="rcvTextOrHex_client"/>
  <buttongroup name="sndTextOrHex_client"/>
 </buttongroups>
</ui>
